include/AST/ASTContext.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef GLU_AST_ASTCONTEXT_HPP
+#define GLU_AST_ASTCONTEXT_HPP
+
+#include "ASTMemoryArena.hpp"
+
+namespace glu::ast {
+
+class ASTContext {
+    ASTMemoryArena _arena;
+
+public:
+    ASTContext() = default;
+
+    /// @brief Get the memory arena used by the AST context.
+    /// @return The memory arena used by the AST context.
+    ASTMemoryArena &getMemoryArena() { return _arena; }
```

Comment:
we'll have two memory arenas in ASTContext: one for ASTNodes, one for Types. The name of the method should reflect which one it is